packages/capacitor-plugin/android/src/main/kotlin/com/capacitorjs/plugins/filesystem/FilesystemMethodOptions.kt

```diff
@@ -0,0 +1,134 @@
+import io.ionic.libs.ionfilesystemlib.model.IONFILEReadOptions
+import io.ionic.libs.ionfilesystemlib.model.IONFILESaveMode
+import io.ionic.libs.ionfilesystemlib.model.IONFILESaveOptions
+import io.ionic.libs.ionfilesystemlib.model.IONFILEUri
+
+internal val INPUT_APPEND = "append"
+private val INPUT_PATH = "path"
+private val INPUT_DIRECTORY = "directory"
+private val INPUT_ENCODING = "encoding"
+private val INPUT_CHUNK_SIZE = "chunkSize"
+private val INPUT_DATA = "data"
+private val INPUT_RECURSIVE = "recursive"
+private val INPUT_FROM = "from"
+private val INPUT_FROM_DIRECTORY = "directory"
+private val INPUT_TO = "to"
+private val INPUT_TO_DIRECTORY = "toDirectory"
+
+internal data class ReadFileOptions(
+    val uri: IONFILEUri.Unresolved,
+    val options: IONFILEReadOptions
+)
+
+internal data class ReadFileInChunksOptions(
+    val uri: IONFILEUri.Unresolved,
+    val options: IONFILEReadInChunksOptions
+)
+
+internal data class WriteFileOptions(
+    val uri: IONFILEUri.Unresolved,
+    val options: IONFILESaveOptions
+)
+
+internal data class SingleUriWithRecursiveOptions(
+    val uri: IONFILEUri.Unresolved,
+    val recursive: Boolean
+)
+
+internal data class DoubleUri(
+    val fromUri: IONFILEUri.Unresolved,
+    val toUri: IONFILEUri.Unresolved,
+)
+
+/**
+ * @return [ReadFileOptions] from JSON inside [PluginCall], or null if input is invalid
+ */
+internal fun PluginCall.getReadFileOptions(): ReadFileOptions? {
+    val uri = getSingleIONFILEUri() ?: return null
+    val encodingName = getString(INPUT_ENCODING)
+    return ReadFileOptions(
+        uri = uri,
+        options = IONFILEReadOptions(IONFILEEncoding.fromEncodingName(encodingName))
+    )
+}
+
+/**
+ * @return [ReadFileInChunksOptions] from JSON inside [PluginCall], or null if input is invalid
+ */
+internal fun PluginCall.getReadFileInChunksOptions(): ReadFileInChunksOptions? {
+    val uri = getSingleIONFILEUri() ?: return null
+    val encodingName = getString(INPUT_ENCODING)
+    val chunkSize = getInt(INPUT_CHUNK_SIZE)?.takeIf { it > 0 } ?: return null
+    return ReadFileInChunksOptions(
+        uri = uri,
+        options = IONFILEReadInChunksOptions(
+            IONFILEEncoding.fromEncodingName(encodingName),
+            chunkSize
+        )
+    )
+}
+
+/**
+ * @return [ReadFileOptions] from JSON inside [PluginCall], or null if input is invalid
+ */
+internal fun PluginCall.getWriteFileOptions(): WriteFileOptions? {
+    val uri = getSingleIONFILEUri() ?: return null
+    val data = getString(INPUT_DATA) ?: return null
+    val recursive = getBoolean(INPUT_RECURSIVE) ?: false
+    val append = getBoolean(INPUT_APPEND) ?: false
+    val encodingName = getString(INPUT_ENCODING)
+    return WriteFileOptions(
+        uri = uri,
+        options = IONFILESaveOptions(
+            data = data,
+            encoding = IONFILEEncoding.fromEncodingName(encodingName),
+            mode = if (append) IONFILESaveMode.APPEND else IONFILESaveMode.WRITE,
```

Comment:
Not super important, but since we're using variables for other parameters like `data`, `uri`, and `createFileRecursive`, maybe we could extract this one into a variable as well. Same for `encoding`